### PR TITLE
Feature for creating Route

### PR DESCRIPTION
**Pull request by Dkaykay**
### Description of your changes

Relates to #657 

Implemented Route 

I have:

- [ x ] Read and followed Crossplane's [contribution process].
- [ x ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

- create route
- delete route
- run make reviewable test

### CODE REVIEW

File: apis/ec2/v1alpha1/generator-config.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+ignore:
+  resource_names:
```
Comment (hasheddan) [inline apis/ec2/v1alpha1/generator-config.yaml:2]: @Dkaykay I think either this file is not being honored or you need to delete and re-run generation as all EC2 AWS resources appear to be added in the `v1alpha1` group here.

**Review by hasheddan — CHANGES_REQUESTED** [review]

File: apis/ec2/v1alpha1/generator-config.yaml (hunk 1 above)
Comment (Dkaykay) [reply apis/ec2/v1alpha1/generator-config.yaml:2]: should be fixed

**Review by AaronME — CHANGES_REQUESTED** [review]
This works as described. Would like to support more selectors on fields that refer to already managed resources.

This branch needs to be brought up-to-date with latest master.

File: package/crds/ec2.aws.crossplane.io_routes.yaml [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+                  natGatewayID:
+                    description: '[IPv4 traffic only] The ID of a NAT gateway.'
+                    type: string
+                  networkInterfaceID:
+                    description: The ID of a network interface.
+                    type: string
+                  region:
+                    description: Region is which region the Route will be created.
+                    type: string
+                  routeTableID:
```
[the remark this answers is not in the document]
Comment (Dkaykay) [reply package/crds/ec2.aws.crossplane.io_routes.yaml:93]: i talked to @hasheddan a while ago, he wanted to support with updating the branch and squashing commits.
He also mentioned the selector features, but currently i dont have capacity to work on that. 
Can we please merge the current state?
When i have capacity again, we can add the selectors

**Comment by AaronME** [comment]
I've taken a pass at fixing up the git history and rebasing. Because of a breaking change caused by [this merge](https://github.com/crossplane/provider-aws/commit/f8b42a22bf07b9ad4eb0de9e1799ea1754dda29f), I rebased up to the last known [good merge](https://github.com/crossplane/provider-aws/commit/009f048d509ba924ce6c456740ab946666a49bb6). The next rebase should be much simpler.

The rename of v1alpha1 to manualv1alpha1 confused git during the rebase. I have made sure that we end up with both v1alpha1 and manualv1alpha1 properly split, which includes removing all VPCCIDRBlock code from v1alpha1.

@Dkaykay, please make sure all your changes are present in [this branch](https://github.com/AaronME/provider-aws/tree/fixup-route-history) and that your tests and code still function as expected.

**Comment by Dkaykay** [comment]
Hi again @AaronME,
thanks for the support!

i checked out your branch and tested to create a route, but currently im getting an error in observe:

```2021-07-29T10:49:59.949+0200    DEBUG   provider-aws    Cannot observe external resource{"controller": "managed/route.ec2.aws.crossplane.io", "request": "/example", "uid": "342f2cfc-56a2-4840-81e9-4f2f072eba27", "version": "15564", "external-name": "example", "error": "failed to describe Route", "errorVerbose": "failed to describe Route\ngithub.com/crossplane/provider-aws/pkg/controller/ec2/route.(*external).Observe\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/pkg/controller/ec2/route/zz_controller.go:89\ngithub.com/crossplane/crossplane-runtime/pkg/reconciler/managed.(*Reconciler).Reconcile\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/github.com/crossplane/crossplane-runtime@v0.14.0/pkg/reconciler/managed/reconciler.go:614\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.0/pkg/internal/controller/controller.go:293\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.0/pkg/internal/controller/controller.go:248\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1.1\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/sigs.k8s.io/controller-runtime@v0.8.0/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext.func1\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.20.1/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.20.1/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.20.1/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.20.1/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.JitterUntilWithContext\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.20.1/pkg/util/wait/wait.go:185\nk8s.io/apimachinery/pkg/util/wait.UntilWithContext\n\t/Users/I541517/github/sap/provider-aws-aaron-eaton/.work/pkg/pkg/mod/k8s.io/apimachinery@v0.20.1/pkg/util/wait/wait.go:99\nruntime.goexit\n\t/usr/local/Cellar/go/1.16.5/libexec/src/runtime/asm_amd64.s:1371"}
2021-07-29T10:49:59.949+0200    DEBUG   controller-runtime.manager.events       Warning {"object": {"kind":"Route","name":"example","uid":"342f2cfc-56a2-4840-81e9-4f2f072eba27","apiVersion":"ec2.aws.crossplane.io/v1alpha1","resourceVersion":"15564"}, "reason": "CannotObserveExternalResource", "message": "failed to describe Route"}```

**Comment by AaronME** [comment]
@Dkaykay I'll also investigate why the changes are causing an observe error. Thanks!

**Comment by nomanuddin** [comment]
Hi @AaronME,

Any success with this error? 

Thank you

**Comment by Dkaykay** [comment]
> @Dkaykay I'll also investigate why the changes are causing an observe error. Thanks!

Any success @AaronME ?

**Comment by Dkaykay** [comment]
Checked again with branch created by @AaronME , not getting observe errors anymore
Maybe the errors were local to my previous test cluster. Today i tested with a new cluster. 
So maybe it will be sufficient to merge master to branch fixup-route-history in https://github.com/AaronME/provider-aws to be ready for merge.

**Comment by AaronME** [comment]
@Dkaykay Thank you for all your hard work on this. I'm going to document the outcomes from our zoom and slack conversations:

- The code in this branch works to create and delete routes.
- Running `make services SERVICE=ec2` results in some unexpected changes, which indicates we need to bring this branch up to date with master and the latest code generation.
- Merging or rebasing on master is causing a large number of conflicts.
- We created https://github.com/aaronme/provider-aws/tree/fixup-route-history in July, but that branch has also fallen well behind master.

I'm working out some git steps to help unblock @dkaykay here.

**Comment by Dkaykay** [comment]
Closing this PR due to a lot of changes. Will recreate the PR from clean branch

**Pull request closed without merging**